packages/dataregistry-csvviewer-extension/src/csv_to_data_grid.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license BSD-3-Clause
+ *
+ * Copyright (c) 2019 Project Jupyter Contributors.
+ * Distributed under the terms of the 3-Clause BSD License.
+ */
+
+import { createConverter } from "@jupyterlab/dataregistry";
+import datatypes from "./datatypes"; // FIXME
+import DataGrid from "./data_grid";
+
+/**
+ * Interface describing an object containing data to convert.
+ *
+ * @private
+ */
+interface Data {
+  /**
+   * Data to convert.
+   */
+  data: any;
+}
+
+/**
+ * Returns a converter for converting CSV data to a data grid.
+ *
+ * @private
+ * @returns data type converter
+ */
+function csv2datagrid(): any {
+  const conversion = {
+    "from": datatypes.csv,
+    "to": datatypes.widget
+  };
+
+  return createConverter(conversion, convert);
```

Comment:
I prefer the functions inline here, since the typescript know the types of everything. Otherwise, you have to manually specify the types. Like below, `convert` has no return annotation, so it is not verified that it is correct. It's a pain to add those annotations so if we have the functions inline, then we dont have to.

Reply:
Yeah, I ended up frustrated with inlining because you cannot readily document inlined functions and cannot elevate them to a higher scope.

I also look at this another way:

> what is the minimum contract necessary for the `convert` function to work and for someone (e.g., an outside developer reading the source code for this extension) to be able to grok ?

Based on how `createConverter` works, I can already assume that `convert` is going to be provided an object containing data of the correct "type", that is part of the `createConverter` "contract", and we've already defined in the `conversion` object the data type in the return object. 

In short, TMK, we don't loose any "correctness" by using general types for what is an internal function interface.

And personally, and this is one of my gripes about using TypeScript, I shouldn't have to choose between precise and exhaustive types and no documentation/scope hoisting at all. TypeScript effectively encourages devs to inline functions (often via arrow functions in order to avoid explicit typing) which is not, IMO, good practice for a variety of reasons (perf, docs, debugging, readability, etc).

This stated, I can modify to the conventions of this project, if this is a blocker for merge. :)

Reply:
Thanks for the explanation here, that makes sense! I haven't thought as much about the downsides of Typescript's strictness, I just let it tell me what to do :)

Let's merge this in.